awx/api/serializers.py

```diff
@@ -4913,7 +4917,19 @@ class Meta:
             'ip_address',
             'listener_port',
         )
-        extra_kwargs = {'node_type': {'initial': 'execution'}, 'node_state': {'initial': 'installed'}}
+        extra_kwargs = {
+            'node_type': {'initial': 'execution'},
+            'node_state': {'initial': 'installed'},
+            'hostname': {
+                'validators': [
+                    MaxLengthValidator(limit_value=255),
+                    HostnameRegexValidator(),
+                    RegexValidator(
+                        regex='^localhost$|^127.0.0.1$', flags=re.IGNORECASE, inverse_match=True, message="hostname cannot be localhost or 127.0.0.1"
```

Comment:
don't we still have to include the IPv6 localhost regex?

Reply:
also I suspect that this one should go 2nd in the list, then the hostname validator